### PR TITLE
Threaded tests for expressions and update version numbers

### PR DESCRIPTION
**Pull request by rok-cesnovar**
#### Summary

This PR adds threaded expression tests and updates version number after the latest release.

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company): Rok Češnovar

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
- Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

### CODE REVIEW

**Comment by rok-cesnovar** [comment]
Ready for review. 

cc: @bbbales2

File: Jenkinsfile [hunk 1]
```diff
@@ -291,6 +291,11 @@ pipeline {
                                 withEnv(['PATH+TBB=./lib/tbb']) {
                                     sh "python ./test/expressions/test_expression_testing_framework.py"
                                 }
+                                sh "make clean-all"
+                                sh "echo STAN_THREADS=true >> make/local"
+                                withEnv(['PATH+TBB=./lib/tbb']) {
+                                    sh "./runTests.py -j${env.PARALLEL} test/expressions"
```
Comment (bbbales2) [inline Jenkinsfile:297]: I see the tests above are wrapped in a try/finally. Something about a junit. Does that need to happen here?

**Review by bbbales2 — CHANGES_REQUESTED** [review]
Review!

File: Jenkinsfile (hunk 1 above)
Comment (rok-cesnovar) [reply Jenkinsfile]: Fixed. Thanks!

**Review by bbbales2 — APPROVED** [review]
Good!

**Pull request merged**